release/models/gnsi/openconfig-gnsi-acctz.yang

```diff
@@ -205,41 +213,44 @@ module openconfig-gnsi-acctz {
     }
   }
 
-  grouping grpc-server-acctz-counters {
+  grouping acctz-counters {
     description
       "A collection of counters from the gNSI.acctz module.";
 
-    container acctz-counters {
+    container counters {
       config false;
       description
         "A collection of counters from the gNSI.acctz module
         for acctz clients and sources.";
 
-      container state {
+      leaf last-cleared {
+        type oc-types:timeticks64;
         description
-          "Operational state relating to acctz-counters.";
-
-        leaf counters-last-cleared {
-          type oc-types:timeticks64;
-          description
-            "The last time that the counters were cleared (reset to
-            zero). This value is reported as nanoseconds since epoch
-            (January 1st, 1970 00:00:00 GMT).";
-        }
-
-        uses client-counters;
+          "The last time that the counters were cleared (reset to
+          zero). This value is reported as nanoseconds since epoch
+          (January 1st, 1970 00:00:00 GMT).";
       }
 
+      uses client-counters;
       uses source-counters;
     }
   }
 
-  // Augments section.
-  augment "/oc-sys:system/oc-sys-grpc:grpc-servers/oc-sys-grpc:grpc-server" {
-    when "config[contains(services, 'oc-gnsi:GNSI')]/enable = 'true'";
+  grouping acctz-base {
     description
-      "Counters collected by the gNSI.acctz module.";
+      "";
+
+    container acctz {
+      description
+        "";
+
+      uses acctz-counters;
+    }
+  }
+
+  augment "/oc-sys:system/oc-sys-grpc:grpc-servers" +
+          "/oc-sys-grpc:grpc-server/oc-sys-grpc:services/oc-gnsi:gnsi" {
 
```

Comment:
Do you want a when statement here? 

Here and throughout the protocol model files, is it simpler to not include any when statement and have the implementation choose whether or not to include the model based on support of the protocol?

Reply:
My thoughts are yes, a when statement should be used here

It is more of a signal that "this subtree should be expected and exists _if_ the service is enabled on the server endpoint"

The model inclusion/exclusion is the hard demarcation where the service is not supported whatsoever